src/utils/wallet.js

```diff
@@ -211,6 +215,74 @@ const wallet = {
     }
   },
 
+  /**
+   * The wallet needs each token metadata to show information correctly
+   * So we fetch the tokens metadata and store on redux
+   *
+   * @param {Array} tokens Array of token uids
+   * @param {String} network Network name
+   * @param {Number} downloadRetry Number of retries already done
+   *
+   * @memberof Wallet
+   * @inner
+   **/
+  async fetchTokensMetadata(tokens, network, downloadRetry = 0) {
+    const metadataPerToken = {};
+    let didBreak = false;
+
+    const handleRetry = (index) => {
+      if (downloadRetry === METADATA_RETRY_LIMIT) {
+        // Should we throw an error and prevent wallet usage only because metadata was not loaded correctly?
+        // Right now I'm continuing and will download as many as we can
+        // Won't break the for
+        return false;
```

Comment:
Should we mark the error in the redux global state? We can use it to show the error on the screens.

Reply:
Added the redux state with the errors but we've decided not to use it for now